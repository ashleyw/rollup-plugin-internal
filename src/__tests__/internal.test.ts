import path from 'path';

import { bundle } from './helpers';
import internal from '..';

it('should have a name', () => {
  expect(internal([]).name).toEqual('internal');
});

it('should do nothing if passed empty array', async () => {
  const result = await bundle({
    config: {
      input: path.resolve(__dirname, '__fixtures__/deps/index.js'),
    },
    modules: [],
  });

  expect(result).toMatchSnapshot();
});

it('should override external array option', async () => {
  const result = await bundle({
    plugins: ['nodeResolve', 'commonjs'],
    config: {
      input: path.resolve(__dirname, '__fixtures__/deps/index.js'),
      external: ['foo'],
    },
    modules: ['foo'],
  });

  expect(result).toMatchSnapshot();
});

it('should override external function option', async () => {
  const result = await bundle({
    plugins: ['nodeResolve', 'commonjs'],
    config: {
      input: path.resolve(__dirname, '__fixtures__/deps/index.js'),
      external: source => source === 'foo',
    },
    modules: ['foo'],
  });

  expect(result).toMatchSnapshot();
});

it('should override autoExternal plugin', async () => {
  const result = await bundle({
    plugins: ['nodeResolve', 'commonjs', 'autoExternal'],
    config: {
      input: path.resolve(__dirname, '__fixtures__/deps/index.js'),
      external: ['baz'],
    },
    modules: ['foo'],
  });

  expect(result).toMatchSnapshot();
});
