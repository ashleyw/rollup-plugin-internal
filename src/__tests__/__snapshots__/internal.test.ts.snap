// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should do nothing if passed empty array 1`] = `
Object {
  "code": "import foo from 'foo';
import bar from 'bar';

var baz = 'baz';

var index = \`\${foo}\${bar}\${baz}\`;

export default index;
",
  "modules": Array [
    "node_modules/baz/baz.js",
    "node_modules/baz/index.js",
  ],
}
`;

exports[`should override autoExternal plugin 1`] = `
Object {
  "code": "import bar from 'bar';
import baz from 'baz';

var foo = 'foo';

var index = \`\${foo}\${bar}\${baz}\`;

export default index;
",
  "modules": Array [
    "node_modules/foo/foo.js",
    "node_modules/foo/index.js",
  ],
}
`;

exports[`should override external array option 1`] = `
Object {
  "code": "var foo = 'foo';

var bar = 'bar';

var bar_1 = bar;

var baz = 'baz';

var index = \`\${foo}\${bar_1}\${baz}\`;

export default index;
",
  "modules": Array [
    "node_modules/foo/foo.js",
    "node_modules/foo/index.js",
    "node_modules/bar/bar.js",
    "node_modules/bar/bar.js?commonjs-proxy",
    "node_modules/bar/index.js",
    "node_modules/baz/baz.js",
    "node_modules/baz/index.js",
  ],
}
`;

exports[`should override external function option 1`] = `
Object {
  "code": "var foo = 'foo';

var bar = 'bar';

var bar_1 = bar;

var baz = 'baz';

var index = \`\${foo}\${bar_1}\${baz}\`;

export default index;
",
  "modules": Array [
    "node_modules/foo/foo.js",
    "node_modules/foo/index.js",
    "node_modules/bar/bar.js",
    "node_modules/bar/bar.js?commonjs-proxy",
    "node_modules/bar/index.js",
    "node_modules/baz/baz.js",
    "node_modules/baz/index.js",
  ],
}
`;
